pkgs/development/tools/sumneko-lua-language-server/default.nix

```diff
@@ -65,6 +65,7 @@ stdenv.mkDerivation rec {
 
     makeWrapper "$out"/share/lua-language-server/bin/lua-language-server \
       $out/bin/lua-language-server \
+      --run 'export XDG_CACHE_HOME="''${XDG_CACHE_HOME:-$HOME/.cache}"' \
```

Comment:
You can use `--set-default` instead of `--run export ...`
```bash
--set-default XDG_CACHE_HOME \$HOME/.cache
```

Reply:
Well that's helpful, thanks :) Using a variable wouldn't work because in the wrapper it ends up in single quotes. `~/.cache` got changed into `/homeless-shelter/.cache`, but `\~/.cache` works as expected.